File: java-example/src/main/java/com/example/javaexample/ctrl/TestCtrl.java
package com.example.javaexample.ctrl;

import com.example.javaexample.ctrl.dto.QueryConstructorDTO;
import com.example.javaexample.ctrl.dto.QueryDTO;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
class TestCtrl {

    @GetMapping(value = "/", produces = {MediaType.APPLICATION_JSON_VALUE})
    public String accountsAsPojo(QueryDTO filter) {
        return "success " + filter.getValues().get(0);
    }
    
    @GetMapping(value = "/ctr", produces = {MediaType.APPLICATION_JSON_VALUE})
    public String accountsAsPojoConstructorBased(QueryConstructorDTO filter) {
        return "success " + filter.getValues().get(0);
    }
    
    @GetMapping(value = "/list", produces = {MediaType.APPLICATION_JSON_VALUE})
    public String accountsListAsParam(@RequestParam List<String> values) {
        return "success " + values.get(0);
    }
}

File: src/main/kotlin/com/example/ktl/ctrl/TestCtrl.kt
package com.example.ktl.ctrl;

import com.example.ktl.ctrl.dto.QueryDTO
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController;

@RestController
class TestCtrl {

    @GetMapping(path = ["/"], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun accounts(filter: QueryDTO): String {
        return "success ${filter.values?.get(0)}"
    }
    
    @GetMapping(path = ["/list"], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun accountsLiast(@RequestParam values: List<String>): String {
        return "success ${values[0]}"
    }
}
